src/ray/raylet/main.cc

```diff
@@ -202,10 +121,112 @@ int main(int argc, char *argv[]) {
   }
   RAY_CHECK_OK(gcs_client->Connect(main_service));
 
-  std::unique_ptr<ray::raylet::Raylet> server(new ray::raylet::Raylet(
-      main_service, raylet_socket_name, node_ip_address, redis_address, redis_port,
-      redis_password, node_manager_config, object_manager_config, gcs_client));
-  server->Start();
+  if (head_node) {
```

Comment:
Maybe leave a comment about why you only do this for head nod

Reply:
Okay!